python/cudf/cudf/_lib/transpose.pyx

```diff
@@ -42,14 +47,16 @@ def transpose(Table source):
     elif any(c.dtype != dtype for c in source._columns):
         raise ValueError('Columns must all have the same dtype')
 
-    cdef unique_ptr[table] c_result
+    cdef pair[unique_ptr[column], table_view] c_result
     cdef table_view c_input = source.data_view()
 
     with nogil:
-        c_result = move(cpp_transpose.transpose(c_input))
+        c_result = move(cpp_transpose(c_input))
 
-    result = Table.from_unique_ptr(
-        move(c_result),
+    result_owner = Column.from_unique_ptr(move(c_result.first))
+    result = Table.from_table_view(
+        c_result.second,
+        owner=result_owner,
         column_names=cudf.core.index.RangeIndex(0, source._num_rows)
```

Comment:
Column names should be a host object and definitely not a GPU object here.

@shwina any idea what's the most efficient way to pass column names of 0 --> N here? As a Python list?

Reply:
I'd imagine `range(0)` if it's suppose to be host-side?